File: python/examples/quickstart.py
"""Quickstart examples."""

from framequery import FrameQuery


def main() -> None:
    fq = FrameQuery(api_key="fq_your_api_key_here")

    # -- Process a local file --
    result = fq.process("interview.mp4")
    print(f"Duration: {result.duration}s")
    print(f"Scenes: {len(result.scenes)}")
    for scene in result.scenes:
        print(f"  [{scene.end_time}s] {scene.description} — {scene.objects}")
    print(f"Transcript segments: {len(result.transcript)}")
    for seg in result.transcript:
        print(f"  [{seg.start_time}-{seg.end_time}s] {seg.text}")

    # -- From URL --
    result = fq.process_url("https://cdn.example.com/video.mp4")
    print(f"Processed URL video: {result.duration}s, {len(result.scenes)} scenes")

    # -- Upload only --
    job = fq.upload("video.mp4")
    print(f"Job created: {job.id} (status: {job.status})")

    job = fq.get_job(job.id)
    print(f"Job status: {job.status}")

    # -- Progress callback --
    def on_progress(job):
        eta = f", ETA: {job.eta_seconds}s" if job.eta_seconds else ""
        print(f"  Status: {job.status}{eta}")

    result = fq.process("video.mp4", on_progress=on_progress)

    # -- Quota --
    quota = fq.get_quota()
    print(f"Plan: {quota.plan}")
    print(f"Credits: {quota.credits_balance_hours}h remaining")
    print(f"Included: {quota.included_hours}h (resets {quota.reset_date})")

    # -- List jobs --
    page = fq.list_jobs(limit=10, status="COMPLETED")
    for job in page.jobs:
        print(f"  {job.id}: {job.status} — {job.filename}")
    if page.has_more:
        next_page = fq.list_jobs(limit=10, cursor=page.next_cursor)
        print(f"  ... and {len(next_page.jobs)} more")


if __name__ == "__main__":
    main()

File: typescript/examples/quickstart.ts
import FrameQuery from "framequery";

async function main() {
  const fq = new FrameQuery({ apiKey: "fq_your_api_key_here" });

  // 1. Process a local file (Node.js — upload + wait)
  const result = await fq.process("./interview.mp4");
  console.log(`Duration: ${result.duration}s`);
  console.log(`Scenes: ${result.scenes.length}`);
  for (const scene of result.scenes) {
    console.log(`  [${scene.endTime}s] ${scene.description} — ${scene.objects.join(", ")}`);
  }
  for (const seg of result.transcript) {
    console.log(`  [${seg.startTime}-${seg.endTime}s] ${seg.text}`);
  }

  // 2. Process from URL
  const urlResult = await fq.processUrl("https://cdn.example.com/video.mp4");
  console.log(`URL video: ${urlResult.duration}s, ${urlResult.scenes.length} scenes`);

  // 3. Upload without waiting
  const job = await fq.upload("./video.mp4");
  console.log(`Job created: ${job.id} (${job.status})`);

  const checked = await fq.getJob(job.id);
  console.log(`Job status: ${checked.status}`);

  // 4. Progress tracking
  const tracked = await fq.process("./video.mp4", {
    onProgress: (j) => {
      const eta = j.etaSeconds ? `, ETA: ${j.etaSeconds}s` : "";
      console.log(`  Status: ${j.status}${eta}`);
    },
  });

  // 5. Check quota
  const quota = await fq.getQuota();
  console.log(`Plan: ${quota.plan}`);
  console.log(`Credits: ${quota.creditsBalanceHours}h remaining`);

  // 6. List jobs
  const page = await fq.listJobs({ limit: 10, status: "COMPLETED" });
  for (const j of page.jobs) {
    console.log(`  ${j.id}: ${j.status} — ${j.filename}`);
  }
  if (page.hasMore) {
    console.log(`  ... more available (cursor: ${page.nextCursor})`);
  }
}

main().catch(console.error);
